Tasktiger

igor47 opened the pull request:
OMFG. Third time is the charm?

I got spammed with a ton of "mission end" emails. turns out this is because of this bug in celery:
https://github.com/celery/kombu/issues/337

which took me a while to even understand, has been open for like 4 years, and nobody has any clue what to do about it. the main suggestion was, switch from redis to rabbitmq, which i am NOT going to do cuz rabbitmq is a PITA.

so, i found yet another worker queue system. wtf, python? this one at least seems pretty simple -- like, the code is all in one repo and i can read it, so maybe if something goes wrong i can fix it. this doesn't have the functionality of expiring tasks if their ETA was too far in the past, but it's not really a hard requirement.

igor47 committed:
rename `celery-worker` to `worker`

allows switching worker queues without renaming stuff

igor47 committed:
replace celery with tasktiger

igor47 committed:
email: don't silently fail on send errors

we might want to capture specific exceptions in specific situations --
like, in test if we are unauthorized

jeredw commented on spaceship/tasktiger.py:
```diff
@@ -0,0 +1,35 @@
+    self.log.debug("disposing of db engine")
+    self.db.engine.dispose()
+
+  def __exit__(self, ex_type, ex_value, ex_tb):
```
should we also call db.session.remove()?

jeredw replied:
(we think this is ok because the connection should get closed when the task subprocess exits)

jeredw commented on requirements.txt:
```diff
@@ -51,15 +47,16 @@ python-http-client==3.1.0
 sendgrid==6.0.4
 six==1.12.0
 SQLAlchemy==1.3.4
+structlog==19.1.0
```
what is structlog?

igor47 committed:
better log config based on root logger

so we're not limiting our logs to the `spaceship` space

igor47 committed:
fix tasktiger logging

for some reason, doing this prevents it from logging debug-level
messages every second.
